### PR TITLE
include/signal.h:Expanding SIGNAL to be consistent with Linux

### PR DESCRIPTION
**Pull request by crafcat7**
## Summary
Expanding SIGNAL to be consistent with Linux

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/signal.h [hunk 1]
```diff
@@ -211,41 +206,96 @@
...
+#  define SIGPROF       27
+#else
+#  define SIGPROF       CONFIG_SIG_PROF
+#endif
+
+#ifndef CONFIG_SIG_POLL
+#  define SIGPOLL       29
+#else
+#  define SIGPOLL       CONFIG_SIG_POLL
 #endif
```
Comment (xiaoxiang781216) [inline include/signal.h:293]: #define SIGIO		SIGPOLL

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by slorquet** [comment]
This was not reviewed before merging, why?

Also this breaks if CONFIG_SCHED_HAS_PARENT is not defined because SIGCHLD will collide with the new values.

**Review by slorquet — COMMENTED** [review]
Careful with CONFIG_SCHED_HAS_PARENT
This code breaks my build because this config is not set.

**Comment by crafcat7** [comment]
> Careful with CONFIG_SCHED_HAS_PARENT
> 
> This code breaks my build because this config is not set.

What is the specific error report? This is the SIGNAL implemented according to https://pubs.opengroup.org/onlinepubs/009695399/basedefs/signal.h.html

**Comment by slorquet** [comment]
duplicate switch case because SIGCHLD ==SIGWORK

when CONFIG_SCHED_HAS_PARENT SIGCHLD is forced to 17

**Comment by crafcat7** [comment]
> 

https://github.com/apache/nuttx/blob/47ce89f757bd7767124fbe0a88cfc8e76e68c147/include/signal.h#L222-L226
https://github.com/apache/nuttx/blob/47ce89f757bd7767124fbe0a88cfc8e76e68c147/include/signal.h#L308-L314
In actual definition SIGCHLD is 17 and SIGWORK is 31.
https://github.com/apache/nuttx/blob/47ce89f757bd7767124fbe0a88cfc8e76e68c147/sched/Kconfig#L1557-L1563
https://github.com/apache/nuttx/blob/47ce89f757bd7767124fbe0a88cfc8e76e68c147/sched/Kconfig#L1660-L1667
And the two values ​​​​are not the same in Kconfig

**Comment by slorquet** [comment]
dont you believe me?
```
CC:  string/lib_strsignal.c string/lib_strsignal.c: In function 'strsignal':
string/lib_strsignal.c:169:7: error: duplicate case value
  169 |       case SIGWORK:
      |       ^~~~
string/lib_strsignal.c:117:7: note: previously used here
  117 |       case SIGCHLD:
      |       ^~~~
```

The problem is that SIGWORK was changed recently from 17 to 31 and my config kept that.

I cant be the only person in this situation.

**Comment by slorquet** [comment]
It was changed here from 20 to 31
https://github.com/apache/nuttx/commit/e6227e19433c4999d500437d0a8a5c05f476ceb1#diff-285d196ad580d55fdaef3fbe54702ee3ddeed0d93cba52a5441d1866e0031726L1584

The comment still shows that it was previously changed from 17 to 20

This is very messy.

**Comment by Donny9** [comment]
> 

hello, what is the value of CONFIG_SIG_CHLD and CONFIG_SIG_SIGWORK  of your board?

**Comment by xiaoxiang781216** [comment]
> dont you believe me?
> 
> ```
> CC:  string/lib_strsignal.c string/lib_strsignal.c: In function 'strsignal':
> string/lib_strsignal.c:169:7: error: duplicate case value
>   169 |       case SIGWORK:
>       |       ^~~~
> string/lib_strsignal.c:117:7: note: previously used here
>   117 |       case SIGCHLD:
>       |       ^~~~
> ```
> 
> The problem is that SIGWORK was changed recently from 17 to 31 and my config kept that.
> 
> I cant be the only person in this situation.

so do you use the expend form of defconfig?

**Comment by slorquet** [comment]
I save the current .config as my defconfig, so it contains all parameters.
This way I am sure that the configuration is stable.

@Donny9 hi, it was stuck at 17 because I keep full configs.

This specific situation is hard to track down.

There is no real way to update a full .config file with updated default values, so this is unavoidable.

TBH this is a minor problem with a complex cause.

**Comment by xiaoxiang781216** [comment]
> I save the current .config as my defconfig, so it contains all parameters. This way I am sure that the configuration is stable.
> 
Ok, that is why the main CI doesn't hit the similar problem since all defconfig in boards save in the normalize form.

**Pull request merged**